packages/common/src/permissions/_internal/checkAssertion.ts

```diff
@@ -169,6 +175,34 @@ function equalityAssertions(
   return response;
 }
 
+/**
+ * Does the propValue "starts-with" or "ends-with" the val?
+ * @param assertion
+ * @param propValue
+ * @param val
+ * @returns
+ */
+function stringAssertions(
+  assertion: IPolicyAssertion,
+  propValue: any, // what we found
+  val: any // what we're looking for
+): PolicyResponse {
+  let response: PolicyResponse = "granted";
+  if (assertion.type === "starts-with" && !propValue.startsWith(val)) {
+    response = "assertion-failed";
+  } else if (
+    assertion.type === "not-starts-with" &&
+    propValue.startsWith(val)
+  ) {
+    response = "assertion-failed";
+  } else if (assertion.type === "ends-with" && !propValue.endsWith(val)) {
+    response = "assertion-failed";
+  } else if (assertion.type === "not-ends-with" && propValue.endsWith(val)) {
+    response = "assertion-failed";
+  }
+  return response;
+}
+
```

Comment:
String starts with, ends with, not starts with, & not ends with assertions. See above for why this was needed.